package/network/utils/wireguard-tools/files/wireguard.init

```diff
@@ -42,4 +42,25 @@ start_service() {
 	mkdir -p "${WG_DIR}"
 	config_load network
 	config_foreach wireguard_check_interface interface
+
+	# delete old peers of related wireguard interface
+	wireguard_check_peers
+}
+
+wireguard_check_peers() {
+	local iface peer
+
+	# get all wireguard peers
+	for peer in $(uci show network | grep =wireguard_); do
+		# extract peer section type
+		peer="${peer##*=}"
+		# extract interface name
+		iface="${peer#*_}"
+
+		# delete peer if iface is not present anymore
+		if ! uci -q show "network.${iface}" 1>/dev/null 2>/dev/null; then
+			uci -q delete "network.@${peer}[-1]"
+			uci commit network
```

Comment:
In case we delete more than one peer, this would commit multiple commits. 

Better move the commit out of this loop and only call it if a peer was deleted.

Reply:
I will remove the said section as requested by jow-